ECMA2Yaml/ECMAHelper/ECMALoader.docs.cs

```diff
@@ -121,6 +123,54 @@ public Docs LoadDocs(XElement dElement)
             };
         }
 
+        /// <summary>Downgrades markdown headers from 1 - 5. So a `#` becomes `##`, but `######` (ie. h6) remains the same.</summary>
+        /// <param name="remarksText">A string of markdown content</param>
+        public static string DowngradeMarkdownHeaders(string remarksText)
+        {
+            if (string.IsNullOrWhiteSpace(remarksText)) return remarksText ?? string.Empty;
+
+            var lines = remarksText.Replace("\r\n", "\n").Split(new[] { '\n' }, StringSplitOptions.None); //handle both unix and windows line endings
+            
+            bool replaceTriggered = false;
+
+            // walk through the content, first adjusting larger headers and moving in reverse
+            for (int headerSize = 5; headerSize > 0; headerSize--)
+                ReplaceTriggered(lines, headerSize, ref replaceTriggered);
+            
+            return replaceTriggered ? string.Join(Environment.NewLine, lines) : remarksText;
+        }
+
+        /// <summary>Modifies the array if a header of the given size is found</summary>
+        private static void ReplaceTriggered(string[] lines, int headerCount, ref bool replaceTriggered)
+        {
+            string headerPrefix = new string('#', headerCount);
+            string newHeaderPrefix = null;
+            bool inCodeFence = false;
+            for (int i = 0; i < lines.Length; i++)
+            {
+                string line = lines[i];
+
+                if (line.StartsWith("```"))
+                    inCodeFence = !inCodeFence;// invert codefence flag
+
+                // this allows for documentation about markdown
+                if (inCodeFence)
+                    continue;
+
+                if (line.StartsWith(headerPrefix) && line.Length > headerCount)
```

Comment:
markdown engines like markdig often allows a few (less than 4) spaces in front of '#'s. Which means the following lines will all render to `<h2>`:
```
## header
 ## header with 1 space
  ## header with 2 spaces
   ## header with 3 spaces
```

Reply:
The code now supports this, along with corresponding unit test